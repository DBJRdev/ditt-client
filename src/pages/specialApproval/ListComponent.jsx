import Immutable from 'immutable';
import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import React from 'react';
import jwt from 'jsonwebtoken';
import {
  Button,
  Modal,
  TextField,
  Table,
} from 'react-ui';
import Layout from '../../components/Layout';
import {
  BUSINESS_TRIP_WORK_LOG,
  HOME_OFFICE_WORK_LOG,
  OVERTIME_WORK_LOG,
  TIME_OFF_WORK_LOG,
  VACATION_WORK_LOG,
} from '../../resources/workMonth';
import { toDayMonthYearFormat } from '../../services/dateTimeService';
import { validateRejectWorkLog } from '../../services/validatorService';
import {
  getStatusLabel,
  getTypeLabel,
} from '../../services/workLogService';
import styles from './specialApproval.scss';

class ListComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      lastApprovedWorkLogId: null,
      lastApprovedWorkLogType: null,
      lastRejectedWorkLogId: null,
      lastRejectedWorkLogType: null,
      rejectWorkLogForm: {
        rejectionMessage: null,
      },
      rejectWorkLogFormValidity: {
        elements: {
          form: null,
          rejectionMessage: null,
        },
        isValid: false,
      },
      showRejectWorkLogForm: false,
      showRejectWorkLogFormId: null,
      showRejectWorkLogFormType: null,
      showWorkLogDetailDialog: false,
      showWorkLogDetailDialogType: null,
    };

    this.changeRejectWorkLogFormHandler = this.changeRejectWorkLogFormHandler.bind(this);
    this.closeDeleteWorkLogForm = this.closeDeleteWorkLogForm.bind(this);
    this.rejectWorkLogHandler = this.rejectWorkLogHandler.bind(this);
    this.closeWorkLogDetail = this.closeWorkLogDetail.bind(this);

    this.formErrorStyle = {
      color: '#a32100',
      textAlign: 'center',
    };
  }

  componentDidMount() {
    if (this.props.token) {
      const decodedToken = jwt.decode(this.props.token);

      if (decodedToken) {
        this.props.fetchSpecialApprovalList(decodedToken.uid);
      }
    }
  }

  getFilteredSpecialApprovals() {
    let specialApprovalList = Immutable.List();

    [
      'businessTripWorkLogs',
      'homeOfficeWorkLogs',
      'overtimeWorkLogs',
      'timeOffWorkLogs',
      'vacationWorkLogs',
    ].forEach((key) => {
      specialApprovalList = specialApprovalList.concat((
        this.props.specialApprovalList.get(key).map((
          workLog => workLog
            .set('rawId', workLog.get('id'))
            .set('id', `${workLog.get('type')}-${workLog.get('id')}`)
        ))
      ));
    });

    specialApprovalList = specialApprovalList.sortBy(workLog => -workLog.get('date'));

    return specialApprovalList;
  }

  handleMarkApproved(id, type) {
    if (this.props.token) {
      const decodedToken = jwt.decode(this.props.token);

      if (decodedToken) {
        let action = null;

        switch (type) {
          case BUSINESS_TRIP_WORK_LOG:
            action = this.props.markBusinessTripWorkLogApproved;
            break;
          case HOME_OFFICE_WORK_LOG:
            action = this.props.markHomeOfficeWorkLogApproved;
            break;
          case OVERTIME_WORK_LOG:
            action = this.props.markOvertimeWorkLogApproved;
            break;
          case TIME_OFF_WORK_LOG:
            action = this.props.markTimeOffWorkLogApproved;
            break;
          case VACATION_WORK_LOG:
            action = this.props.markVacationWorkLogApproved;
            break;
          default:
            throw new Error(`Unknown type ${type}`);
        }

        this.setState({
          lastApprovedWorkLogId: id,
          lastApprovedWorkLogType: type,
          lastRejectedWorkLogId: null,
          lastRejectedWorkLogType: null,
        });

        return action(id).then((response) => {
          this.props.fetchSpecialApprovalList(decodedToken.uid);

          return response;
        });
      }
    }

    return null;
  }

  handleMarkRejected(id, type, rejectionMessage) {
    if (this.props.token) {
      const decodedToken = jwt.decode(this.props.token);

      if (decodedToken) {
        let action = null;

        switch (type) {
          case BUSINESS_TRIP_WORK_LOG:
            action = this.props.markBusinessTripWorkLogRejected;
            break;
          case HOME_OFFICE_WORK_LOG:
            action = this.props.markHomeOfficeWorkLogRejected;
            break;
          case OVERTIME_WORK_LOG:
            action = this.props.markOvertimeWorkLogRejected;
            break;
          case TIME_OFF_WORK_LOG:
            action = this.props.markTimeOffWorkLogRejected;
            break;
          case VACATION_WORK_LOG:
            action = this.props.markVacationWorkLogRejected;
            break;
          default:
            throw new Error(`Unknown type ${type}`);
        }

        this.setState({
          lastApprovedWorkLogId: null,
          lastApprovedWorkLogType: null,
          lastRejectedWorkLogId: id,
          lastRejectedWorkLogType: type,
        });

        return action(id, { rejectionMessage }).then((response) => {
          this.props.fetchSpecialApprovalList(decodedToken.uid);

          return response;
        });
      }
    }

    return null;
  }

  openRejectWorkLogForm(id, type) {
    this.setState({
      showRejectWorkLogForm: true,
      showRejectWorkLogFormId: id,
      showRejectWorkLogFormType: type,
    });
  }

  closeDeleteWorkLogForm() {
    this.setState({
      rejectWorkLogForm: {
        rejectionMessage: null,
      },
      showRejectWorkLogForm: false,
      showRejectWorkLogFormId: null,
      showRejectWorkLogFormType: null,
    });
  }

  changeRejectWorkLogFormHandler(e) {
    const eventTarget = e.target;

    this.setState((prevState) => {
      const rejectWorkLogForm = Object.assign({}, prevState.rejectWorkLogForm);
      rejectWorkLogForm[eventTarget.id] = eventTarget.value;

      return { rejectWorkLogForm };
    });
  }

  openWorkLogDetail(id, type) {
    if (BUSINESS_TRIP_WORK_LOG === type) {
      this.props.fetchBusinessTripWorkLog(id);
    }

    this.setState({
      showWorkLogDetailDialog: true,
      showWorkLogDetailDialogType: type,
    });
  }

  closeWorkLogDetail() {
    this.setState({
      showWorkLogDetailDialog: false,
      showWorkLogDetailDialogType: null,
    });
  }

  rejectWorkLogHandler() {
    const {
      rejectWorkLogForm,
      showRejectWorkLogFormId,
      showRejectWorkLogFormType,
    } = this.state;
    const rejectWorkLogFormValidity = validateRejectWorkLog(rejectWorkLogForm);

    this.setState({ rejectWorkLogFormValidity });

    if (rejectWorkLogFormValidity.isValid) {
      this.handleMarkRejected(
        showRejectWorkLogFormId,
        showRejectWorkLogFormType,
        rejectWorkLogForm.rejectionMessage
      )
        .then((response) => {
          if (response.type.endsWith('SUCCESS')) {
            this.closeDeleteWorkLogForm();
          } else if (response.type.endsWith('FAILURE')) {
            rejectWorkLogFormValidity.elements.form = 'Work log cannot be rejected.';

            this.setState({ rejectWorkLogFormValidity });
          }
        });
    }
  }

  renderWorkLogForm() {
    return (
      <Modal
        actions={[
          {
            clickHandler: this.rejectWorkLogHandler,
            label: 'Reject',
            loading: this.props.isPosting,
          },
        ]}
        closeHandler={this.closeDeleteWorkLogForm}
        title="Reject"
      >
        <form>
          <p style={this.formErrorStyle}>
            {this.state.rejectWorkLogFormValidity.elements.form}
          </p>
          <p>Are you sure your want to reject this application?</p>
          <TextField
            changeHandler={this.changeRejectWorkLogFormHandler}
            error={this.state.rejectWorkLogFormValidity.elements.rejectionMessage}
            fieldId="rejectionMessage"
            label="Reason"
            value={this.state.rejectWorkLogForm.rejectionMessage || ''}
          />
        </form>
      </Modal>
    );
  }

  renderWorkLogDetail() {
    let content = 'Loadingâ€¦';

    if (
      BUSINESS_TRIP_WORK_LOG === this.state.showWorkLogDetailDialogType
      && this.props.businessTripWorkLog
    ) {
      content = (
        <p>
          Date: {toDayMonthYearFormat(this.props.businessTripWorkLog.get('date'))}<br />
          Status: {getStatusLabel(this.props.businessTripWorkLog.get('status'))}<br />
          Purpose: {this.props.businessTripWorkLog.get('purpose')}<br />
          Destination: {this.props.businessTripWorkLog.get('destination')}<br />
          Transport: {this.props.businessTripWorkLog.get('transport')}<br />
          Expected departure: {this.props.businessTripWorkLog.get('expectedDeparture')}<br />
          Expected arrival: {this.props.businessTripWorkLog.get('expectedArrival')}
        </p>
      );
    }

    return (
      <Modal
        actions={[]}
        closeHandler={this.closeWorkLogDetail}
        title={getTypeLabel(this.state.showWorkLogDetailDialogType)}
      >
        {content}
      </Modal>
    );
  }

  render() {
    const specialApprovals = this.getFilteredSpecialApprovals();

    return (
      <Layout title="Special approvals" loading={this.props.isFetching}>
        {specialApprovals.count() > 0 ? (
          <Table
            columns={[
              {
                format: row => `${row.workMonth.user.firstName} ${row.workMonth.user.lastName}`,
                label: 'Name',
                name: 'lastName',
              },
              {
                format: row => toDayMonthYearFormat(row.date),
                label: 'Date',
                name: 'date',
              },
              {
                format: row => getTypeLabel(row.type),
                label: 'Type',
                name: 'type',
              },
              {
                format: row => (
                  <div>
                    <div className={styles.workLogButtonWrapper}>
                      <Button
                        clickHandler={() => this.handleMarkApproved(row.rawId, row.type)}
                        label="Approve"
                        loading={
                          this.props.isPosting
                          && this.state.lastApprovedWorkLogId === row.rawId
                          && this.state.lastApprovedWorkLogType === row.type
                        }
                        priority="default"
                        variant="success"
                      />
                    </div>
                    <div className={styles.workLogButtonWrapper}>
                      <Button
                        clickHandler={() => this.openRejectWorkLogForm(row.rawId, row.type)}
                        label="Reject"
                        loading={
                          this.props.isPosting
                          && this.state.lastRejectedWorkLogId === row.rawId
                          && this.state.lastRejectedWorkLogType === row.type
                        }
                        priority="default"
                        variant="danger"
                      />
                    </div>
                    {BUSINESS_TRIP_WORK_LOG === row.type && (
                      <div className={styles.workLogButtonWrapper}>
                        <Button
                          clickHandler={() => this.openWorkLogDetail(row.rawId, row.type)}
                          label="Detail"
                          priority="default"
                        />
                      </div>
                    )}
                  </div>
                ),
                label: 'Actions',
                name: 'actions',
              },
            ]}
            rows={specialApprovals.toJS()}
          />
        ) : (
          <div>
            You do not seem to have any pending special approvals.
          </div>
        )}
        {this.state.showRejectWorkLogForm ? this.renderWorkLogForm() : null}
        {this.state.showWorkLogDetailDialog ? this.renderWorkLogDetail() : null}
      </Layout>
    );
  }
}

ListComponent.defaultProps = {
  businessTripWorkLog: null,
};

ListComponent.propTypes = {
  businessTripWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    destination: PropTypes.string.isRequired,
    expectedArrival: PropTypes.string.isRequired,
    expectedDeparture: PropTypes.string.isRequired,
    purpose: PropTypes.string.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
    transport: PropTypes.string.isRequired,
  }),
  fetchBusinessTripWorkLog: PropTypes.func.isRequired,
  fetchSpecialApprovalList: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  isPosting: PropTypes.bool.isRequired,
  markBusinessTripWorkLogApproved: PropTypes.func.isRequired,
  markBusinessTripWorkLogRejected: PropTypes.func.isRequired,
  markHomeOfficeWorkLogApproved: PropTypes.func.isRequired,
  markHomeOfficeWorkLogRejected: PropTypes.func.isRequired,
  markOvertimeWorkLogApproved: PropTypes.func.isRequired,
  markOvertimeWorkLogRejected: PropTypes.func.isRequired,
  markTimeOffWorkLogApproved: PropTypes.func.isRequired,
  markTimeOffWorkLogRejected: PropTypes.func.isRequired,
  markVacationWorkLogApproved: PropTypes.func.isRequired,
  markVacationWorkLogRejected: PropTypes.func.isRequired,
  specialApprovalList: ImmutablePropTypes.mapContains({
    businessTripWorkLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      date: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      type: PropTypes.oneOf([BUSINESS_TRIP_WORK_LOG]).isRequired,
      workMonth: ImmutablePropTypes.mapContains({
        user: ImmutablePropTypes.mapContains({
          firstName: PropTypes.string.isRequired,
          lastName: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    })).isRequired,
    homeOfficeWorkLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      date: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      type: PropTypes.oneOf([HOME_OFFICE_WORK_LOG]).isRequired,
      workMonth: ImmutablePropTypes.mapContains({
        user: ImmutablePropTypes.mapContains({
          firstName: PropTypes.string.isRequired,
          lastName: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    })).isRequired,
    overtimeWorkLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      date: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      type: PropTypes.oneOf([OVERTIME_WORK_LOG]).isRequired,
      workMonth: ImmutablePropTypes.mapContains({
        user: ImmutablePropTypes.mapContains({
          firstName: PropTypes.string.isRequired,
          lastName: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    })).isRequired,
    timeOffWorkLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      date: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      type: PropTypes.oneOf([TIME_OFF_WORK_LOG]).isRequired,
      workMonth: ImmutablePropTypes.mapContains({
        user: ImmutablePropTypes.mapContains({
          firstName: PropTypes.string.isRequired,
          lastName: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    })).isRequired,
    vacationWorkLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      date: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      type: PropTypes.oneOf([VACATION_WORK_LOG]).isRequired,
      workMonth: ImmutablePropTypes.mapContains({
        user: ImmutablePropTypes.mapContains({
          firstName: PropTypes.string.isRequired,
          lastName: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    })).isRequired,
  }).isRequired,
  token: PropTypes.string.isRequired,
};

export default ListComponent;
