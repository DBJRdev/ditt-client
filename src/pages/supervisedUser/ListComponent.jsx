import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import React from 'react';
import jwt from 'jsonwebtoken';
import { Table } from 'react-ui';
import Layout from '../../components/Layout';

class ListComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      tableSortColumn: 'lastName',
      tableSortDirection: 'asc',
    };
  }

  componentDidMount() {
    if (this.props.token) {
      const decodedToken = jwt.decode(this.props.token);

      if (decodedToken) {
        this.props.fetchSupervisedUserList(
          decodedToken.uid,
          {
            order: {
              column: this.state.tableSortColumn,
              direction: this.state.tableSortDirection,
            },
          }
        );
      }
    }
  }

  render() {
    return (
      <Layout title="Supervised users" loading={this.props.isFetching}>
        <Table
          columns={[
            {
              format: row => `${row.firstName} ${row.lastName}`,
              isSortable: true,
              label: 'Name',
              name: 'lastName',
            },
            {
              format: () => '-',
              label: 'Need approval',
              name: 'needApproval',
            },
            {
              format: () => '-',
              label: 'Show worklog',
              name: 'showWorklog',
            },
          ]}
          rows={this.props.supervisedUserList.toJS()}
          sort={{
            changeHandler: (column, direction) => {
              if (this.props.token) {
                const decodedToken = jwt.decode(this.props.token);

                if (decodedToken) {
                  const orderDirection = direction === 'asc' ? 'desc' : 'asc';

                  this.props.fetchSupervisedUserList(
                    decodedToken.uid,
                    {
                      order: {
                        column,
                        direction: orderDirection,
                      },
                    }
                  ).then(() => {
                    this.setState({
                      tableSortColumn: column,
                      tableSortDirection: orderDirection,
                    });
                  });
                }
              }
            },
            column: this.state.tableSortColumn,
            direction: this.state.tableSortDirection,
          }}
        />
      </Layout>
    );
  }
}

ListComponent.propTypes = {
  fetchSupervisedUserList: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  supervisedUserList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    firstName: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    lastName: PropTypes.string.isRequired,
    supervisor: ImmutablePropTypes.mapContains({
      firstName: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      lastName: PropTypes.string.isRequired,
    }),
  })).isRequired,
  token: PropTypes.string.isRequired,
};

export default ListComponent;
