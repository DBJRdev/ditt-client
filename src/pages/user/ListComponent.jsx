import moment from 'moment-timezone';
import jwt from 'jsonwebtoken';
import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import React from 'react';
import {
  Button,
  Table,
} from 'react-ui';
import { Link } from 'react-router-dom';
import routes from '../../routes';
import Layout from '../../components/Layout';
import { ROLE_SUPER_ADMIN } from '../../resources/user';
import styles from './user.scss';

class ListComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      tableSortColumn: 'lastName',
      tableSortDirection: 'asc',
    };
  }

  componentDidMount() {
    this.props.fetchUserList({
      order: {
        column: this.state.tableSortColumn,
        direction: this.state.tableSortDirection,
      },
    });
  }

  render() {
    const year = moment().year();
    const columns = [
      {
        format: row => (
          /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
          <Link to={routes.editUser.replace(':id', row.id)}>
            {row.firstName} {row.lastName}
          </Link>
        ),
        isSortable: true,
        label: 'Name',
        name: 'lastName',
      },
      {
        format: (row) => {
          if (row.supervisor) {
            return `${row.supervisor.firstName} ${row.supervisor.lastName}`;
          }

          return '-';
        },
        isSortable: true,
        label: 'Supervisor',
        name: 'supervisor.lastName',
      },
      {
        format: (row) => {
          const userYearStats = row.yearStats.filter(stats => stats.year === year)[0];

          return `${Math.round(userYearStats.workedHours * 10) / 10}/${userYearStats.requiredHours}`;
        },
        isSortable: false,
        label: 'Worked / Required hours',
        name: 'requiredWorkedHours',
      },
      {
        format: row =>
          `${row.yearStats.filter(stats => stats.year === year)[0].vacationDaysUsed}/${row.vacationDays}`,
        isSortable: false,
        label: 'Vacation days used',
        name: 'vacationDaysUsed',
      },
    ];

    if (this.props.token) {
      const decodedToken = jwt.decode(this.props.token);

      if (decodedToken && decodedToken.roles.some(role => ROLE_SUPER_ADMIN === role)) {
        columns.push({
          format: row => (
            /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
            <Link to={routes.supervisedUserWorkLog.replace(':id', row.id)}>
              Show work log
            </Link>
          ),
          label: 'Show work log',
          name: 'showWorkLog',
        });
      }
    }

    return (
      <Layout title="Users" loading={this.props.isFetching}>
        <div className={styles.actions}>
          <Button
            clickHandler={() => this.props.history.push(routes.addUser)}
            label="Add user"
            priority="primary"
          />
        </div>
        <Table
          columns={columns}
          rows={this.props.userList.toJS()}
          sort={{
            changeHandler: (column, direction) => {
              const orderDirection = direction === 'asc' ? 'desc' : 'asc';

              this.props.fetchUserList({
                order: {
                  column,
                  direction: orderDirection,
                },
              }).then(() => {
                this.setState({
                  tableSortColumn: column,
                  tableSortDirection: orderDirection,
                });
              });
            },
            column: this.state.tableSortColumn,
            direction: this.state.tableSortDirection,
          }}
        />
      </Layout>
    );
  }
}

ListComponent.propTypes = {
  fetchUserList: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  isFetching: PropTypes.bool.isRequired,
  token: PropTypes.string.isRequired,
  userList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    firstName: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    lastName: PropTypes.string.isRequired,
    supervisor: ImmutablePropTypes.mapContains({
      firstName: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      lastName: PropTypes.string.isRequired,
    }),
  })).isRequired,
};

export default ListComponent;
