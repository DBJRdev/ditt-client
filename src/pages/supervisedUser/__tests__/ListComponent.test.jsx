import Immutable from 'immutable';
import React from 'react';
import { shallow } from 'enzyme';
import ListComponent from '../ListComponent';

describe('rendering', () => {
  it('renders correctly', () => {
    const props = {
      fetchSupervisedUserList: () => {},
      history: {
        push: () => {},
      },
      isFetching: false,
      supervisedUserList: Immutable.List([]),
      token: 'token',
    };
    const tree = shallow(<ListComponent {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
