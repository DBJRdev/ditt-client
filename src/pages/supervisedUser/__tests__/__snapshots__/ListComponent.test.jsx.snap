// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<Connect(LayoutComponent)
  loading={false}
  title="Supervised users"
>
  <Table
    columns={
      Array [
        Object {
          "format": [Function],
          "isSortable": true,
          "label": "Name",
          "name": "lastName",
        },
        Object {
          "format": [Function],
          "label": "Need approval",
          "name": "needApproval",
        },
        Object {
          "format": [Function],
          "label": "Show worklog",
          "name": "showWorklog",
        },
      ]
    }
    rows={Array []}
    sort={
      Object {
        "changeHandler": [Function],
        "column": "lastName",
        "direction": "asc",
      }
    }
  />
</Connect(LayoutComponent)>
`;
